tools/ports/__init__.py

```diff
@@ -577,9 +578,11 @@ def add_cflags(args, settings): # noqa: U100
   needed = get_needed_ports(settings)
 
   # Now get (i.e. build) the ports in dependency order.  This is important because the
-  # headers from one ports might be needed before we can build the next.
+  # headers from one port might be needed before we can build the next.
   for port in dependency_order(needed):
-    port.get(Ports, settings, shared)
+    # When using embuilder, don't build the dependencies
+    if not os.getenv('EMBUILDER_PORT_BUILD_DEFERRED'):
+      port.get(Ports, settings, shared)
```

Comment:
I'm not sure this will work either since what if I do `emcc -sUSE_SDL=2 -c foo.c`.. in this case I would want at least the headers to be installed at this point/

Reply:
Oh I see, in that case `EMBUILDER_PORT_BUILD_DEFERRED` would never be set...